core/drivers/xiphera_trng.c

```diff
@@ -85,6 +82,12 @@ TEE_Result hw_get_random_bytes(void *buf, size_t len)
 	while (buffer_pos < len) {
 		uint32_t exceptions = cpu_spin_lock_xsave(&trng_lock);
 
+		/* Wait until we have value available */
+		if (!xiphera_trng_random_available()) {
```

Comment:
This check is only needed if `xiphera_trng_read32()` is going to be called below.

Reply:
Right. Let's see what can be done for that.

Reply:
Addressed.